modules/features/discover/src/main/java/au/com/shiftyjelly/pocketcasts/discover/view/MostPopularPodcastsAdapter.kt

```diff
@@ -85,20 +84,25 @@ internal class MostPopularPodcastsAdapter(
             },
         )
     }
-    override fun onBindViewHolder(holder: MostPopularPodcastsViewHolder, position: Int) {
-        val podcast = getItem(position)
 
-        if (podcast is DiscoverPodcast) {
-            holder.bind(podcast)
-        }
+    override fun onBindViewHolder(holder: MostPopularPodcastsViewHolder, position: Int) {
+        holder.bind(getItem(position))
     }
+
     fun replaceList(list: List<DiscoverPodcast>) {
-        submitList(null) // We need this to avoid displaying the previous category list when switching from a different category
+        val changedPodcastList = currentList.map { it.uuid } != list.map { it.uuid }
```

Comment:
I tried to use `PODCASTS_FILTERED_DIFF` instead of checking here, but it did not work